mlir/lib/Gradient/IR/GradientOps.cpp

```diff
@@ -54,6 +55,15 @@ LogicalResult verifyGradInputs(OpState *op_state, func::FuncOp callee, ValueRang
         return op_state->emitOpError("incorrect number of operands for callee, ")
                << "expected " << fnType.getNumInputs() << " but got " << fnArgs.size();
 
+    if (callee->getAttrOfType<UnitAttr>("catalyst.invalidGradientOperation")) {
```

Comment:
Is there a way to register this attribute somewhere in our dialect? e.g. [here](https://github.com/PennyLaneAI/catalyst/blob/663de3c89c25c7d949ce0f8fea9578c15abcd1cf/mlir/include/Quantum/IR/QuantumOps.td#L64)

Reply:
Yes! Potentially. I am not an expert yet on using MLIR's libraries. :sweat_smile: I'll give it a try.